drivers/scd30/scd30_saul.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2020 Technische Universität Braunschweig
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_scd30
+ * @file
+ * @brief       SAUL adaption for Sensirion SCD30 sensor
+ * @author      Puhang Ding      <czarsir@gmail.com>
+ * @author		Nishchay Agrawal <f2016088@pilani.bits-pilani.ac.in>
+ * @author      Jan Schlichter   <schlichter@ibr.cs.tu-bs.de>
+ */
+
+#include "saul.h"
+#include "scd30.h"
+#include "scd30_internal.h"
+
+static void _float_fit(float *src, phydat_t *data, uint32_t mul)
```

Comment:
Why is `src` a pointer?

Reply:
No particular reason, but it doesn't really make a difference in this

Reply:
Yea it's not really idiomatic.
Unless you want to modify the content, single values are usually passed by value and not by pointer.
So when reading the function declaration, one would assume that it modifies `src`, but it doesn't.

It's easier to understand the code and spot errors if it follows some common patterns, so when it does something 'unusual' that should be justified.